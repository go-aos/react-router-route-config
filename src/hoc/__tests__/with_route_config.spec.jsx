import { RouteConfig } from 'route-config'
import { string } from 'prop-types'
import React from 'react'
import TestRenderer from 'react-test-renderer'

import Provider from '../../components/provider'
import withRouteConfig from '../with_route_config'

class HasRouteConfig extends React.Component {
  static displayName = 'HasRouteConfig'
  static propTypes = {
    name: string
  }
  static defaultProps = {
    name: 'routeConfig'
  }
  render () {
    return this.props[this.props.name] && <span>Has Route Config ;)</span>
  }
}
const routeConfig = new RouteConfig({
  routes: {
    home: { path: '/home' }
  }
})
const createComp = name => {
  const Component = withRouteConfig(name)(HasRouteConfig)
  return TestRenderer.create(
    <Provider routeConfig={routeConfig}>
      <Component name={name} />
    </Provider>
  )
}

describe('withRouteConfig hoc', () => {
  test('default prop name', () => {
    const comp = createComp()
    expect(comp.root.findByType(HasRouteConfig).props.routeConfig)
      .toBe(routeConfig)
    expect(comp.toJSON()).toMatchSnapshot()
  })

  test('custom prop name', () => {
    const comp = createComp('map')
    expect(comp.root.findByType(HasRouteConfig).props.map).toBe(routeConfig)
    expect(comp.toJSON()).toMatchSnapshot()
  })
})
