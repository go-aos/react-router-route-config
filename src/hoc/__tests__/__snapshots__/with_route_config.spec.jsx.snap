// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`withRouteConfig hoc custom prop name 1`] = `
<span>
  Has Route Config ;)
</span>
`;

exports[`withRouteConfig hoc default prop name 1`] = `
<span>
  Has Route Config ;)
</span>
`;
